From 003c1d8d0f91885474c872db38b750a091007ab5 Mon Sep 17 00:00:00 2001
From: Nicolas Boulenguez <nicolas.boulenguez@free.fr>
Date: Sun, 31 Jul 2016 00:25:12 +0200
Subject: [PATCH 08/12] Various style improvements: * Standards-Version: 3.9.8 
  Explain status of init script in extended description. * Update homepage. *
 Rely on dpkg-dev to set DEB_HOST_MULTIARCH efficiently. * Link with
 --as-needed to reduce actual dependencies on some architectures. * Declare
 phony targets in debian/rules. * Explicitly disable aacplus to ensure
 reproducible builds   even if the library is available on the build system.

---
 debian/control | 11 +++++++++--
 debian/rules   | 22 ++++++++++++++++++++--
 2 files changed, 29 insertions(+), 4 deletions(-)

diff --git a/debian/control b/debian/control
index 6b9aaa9..e20c003 100644
--- a/debian/control
+++ b/debian/control
@@ -2,9 +2,11 @@ Source: darkice
 Section: sound
 Priority: optional
 Maintainer: Jochen Friedrich <jochen@scram.de>
-Standards-Version: 3.9.2
+Standards-Version: 3.9.8
 Build-Depends: debhelper (>= 9~), libvorbis-dev, libasound2-dev [!kfreebsd-i386 !kfreebsd-amd64 !hurd-i386], libjack-dev, libtwolame-dev, dh-autoreconf, libpulse-dev, libsamplerate0-dev, libopus-dev, libmp3lame-dev
-Homepage: http://code.google.com/p/darkice/
+# dpgk-dev provides /usr/share/dpkg/architecture.mk.
+ ,dpkg-dev (>= 1.16.1),
+Homepage: http://www.darkice.org/
 
 Package: darkice
 Architecture: any
@@ -17,3 +19,8 @@ Description: Live audio streamer
  ShoutCast servers, the Ogg Vorbis stream to one or more IceCast2
  servers. DarkIce uses lame as a shared object as its mp3 encoder, and
  the Ogg Vorbis as its Ogg Vorbis encoder.
+ .
+ A legacy SYSV init script is provided for existing
+ configurations. You have to set RUN=yes in /etc/default/darkice to
+ run darkice as a daemon. With modern systems, this is not needed as
+ realtime scheduling is available to normal users.
diff --git a/debian/rules b/debian/rules
index ec53039..9c57cbf 100755
--- a/debian/rules
+++ b/debian/rules
@@ -1,10 +1,27 @@
 #!/usr/bin/make -f
 
-DEB_HOST_MULTIARCH ?= $(shell dpkg-architecture -qDEB_HOST_MULTIARCH)
+include /usr/share/dpkg/architecture.mk
 
-%:
+# Dependency on NSL is not really required on some architectures.
+export DEB_LDFLAGS_MAINT_APPEND := -Wl,--as-needed -Wl,-z,defs
+export DEB_BUILD_MAINT_OPTIONS := hardening=+all
+
+POLICY_TARGETS := binary binary-arch binary-indep build build-arch	\
+build-indep clean
+.PHONY: $(POLICY_TARGETS)
+$(POLICY_TARGETS):
 	dh $@ --with autoreconf
 
+# lame lacks pkg-config support and requires an explicit option.
+
+# vorbis, opus, twolame, alsa, pulseaudio, jack, samplerate are listed
+# in Build-Depends and selected automatically if present.
+
+# faac, aacplus are not in Debian main. Explicitly disable them so
+# that the build does not change if they are installed or removed.
+
+# Trick sysconfdir to install the examples in the right place.
+.PHONY: override_dh_auto_configure
 override_dh_auto_configure:
 	dh_auto_configure -- \
 		--with-lame-prefix=/usr/lib/$(DEB_HOST_MULTIARCH) \
@@ -16,4 +33,5 @@ override_dh_auto_configure:
 		--with-jack-prefix=/usr/lib/$(DEB_HOST_MULTIARCH) \
 		--prefix=/usr \
 		--sysconfdir=/usr/share/doc/darkice/examples \
+		--without-aacplus \
 		--without-faac
-- 
2.8.1

